controllers/cloud.redhat.com/providers/database/appinterface.go

```diff
@@ -149,11 +149,12 @@ func GetDbConfig(ctx context.Context, pClient client.Client, namespace, appName,
 		matched = resolveDb(dbSpec, dbConfigs)
 
 		if matched == (config.DatabaseConfigContainer{}) {
-			return nil, &errors.MissingDependencies{
-				MissingDeps: map[string][]string{
-					"database": {appName},
-				},
-			}
+			missingDep := errors.MakeMissingDependencies(errors.MissingDependency{
+				Source:  "database",
+				App:     appName,
+				Details: "No app interface database provider config found",
```

Comment:
This can probably be addressed in a follow-up -- technically I think we could go deeper on the details here... If you look at `genDbConfigs` and `resolveDb` we are looking for a specific secret in the namespace that has specific details in there. We should be able to say something like `DB secret named '%s' not found in namespace '%s'` or something like that -- but need to read the logic of those functions more closely to determine what to sub in for the strings in that message :)